src/System/include/BipedalLocomotion/System/ForwardEuler.h

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file ForwardEuler.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_SYSTEM_FORWARD_EULER_H
+#define BIPEDAL_LOCOMOTION_SYSTEM_FORWARD_EULER_H
+
+#include <tuple>
+
+#include <iDynTree/Core/EigenHelpers.h>
+
+#include <BipedalLocomotion/System/FixStepIntegrator.h>
+
+namespace BipedalLocomotion
+{
+namespace System
+{
+
+/**
+ * Forward Euler integration method.
+ * @tparam DynamicalSystemDerived a class derived from DynamicalSystem
+ * @warning We assume that the vectors / matrices contained in the DynamicalSystemDerived::StateType
+ * and DynamicalSystemDerived::StateDerivativeType can be converted into an Eigen::Map using
+ * iDynTree::toEigen() function
+ */
+template <typename DynamicalSystemDerived>
+class ForwardEuler : public FixStepIntegrator<DynamicalSystemDerived>
+{
+    typename DynamicalSystemDerived::InputType m_computationalBuffer;
+
+    template <std::size_t I = 0, typename... Tp, typename... Td>
+    inline typename std::enable_if<I == sizeof...(Tp), void>::type
+    addArea(const std::tuple<Tp...>& dx, const double& dT, std::tuple<Td...>& x)
+    {
+        static_assert(sizeof...(Tp) == sizeof...(Td));
+    }
+
+    template <std::size_t I = 0, typename... Tp, typename... Td>
+        inline typename std::enable_if < I<sizeof...(Tp), void>::type
+        addArea(const std::tuple<Tp...>& dx, const double& dT, std::tuple<Td...>& x)
+    {
+        static_assert(sizeof...(Tp) == sizeof...(Td));
+
+        iDynTree::toEigen(std::get<I>(x)) += iDynTree::toEigen(std::get<I>(dx)) * dT;
+        addArea<I + 1>(dx, dT, x);
+    }
```

Comment:
What is the point of this functions? Why they are called `addArea` ?

Reply:
I think I got it know, it is to iterate over the elements of the tuple at compile time. Probably a short one line comment or a most descriptive name would help.

Reply:
@GiulioRomualdi I am still curious about why the method is called `addArea`.

Reply:
I'm sure that we can improve the name. The method is named `addArea` because it adds the area under the curve to compute the integral
![image](https://user-images.githubusercontent.com/16744101/85017564-f2b84100-b16b-11ea-9ebf-d874d9fc24ca.png)

Since I don't like this name, I'm open to suggestions

Reply:
Ah, I got it now.